core/src/main/java/org/mule/lifecycle/RegistryLifecycleManager.java

```diff
@@ -32,9 +34,12 @@
 
 public class RegistryLifecycleManager extends AbstractLifecycleManager<Registry> implements RegistryLifecycleHelpers
 {
-    protected Map<String, LifecyclePhase> phases = new HashMap<String, LifecyclePhase>();
-    protected TreeMap<String, LifecycleCallback> callbacks = new TreeMap<String, LifecycleCallback>();
+
+
+    protected Map<String, LifecyclePhase> phases = new HashMap<>();
+    protected TreeMap<String, LifecycleCallback> callbacks = new TreeMap<>();
     protected MuleContext muleContext;
+    private final LifecycleInterceptor lifecycleInterceptor = new InitDisposeLifecycleInterceptor();
```

Comment:
since this attribute is only used for init-dispose, its name should reflect that.